addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_cache.py

```diff
@@ -37,8 +37,20 @@ def wrapper_cached(*args, **kwargs):
             export_settings = args[-1]
             cache_key_args = args[:-1]
 
+        __by_name = ['Object', 'Scene', 'Material', 'Action', 'Mesh']
+
         # we make a tuple from the function arguments so that they can be used as a key to the cache
-        cache_key = tuple(cache_key_args + tuple(cache_key_kwargs.values()))
+        cache_key = ()
+        for i in cache_key_args:
+            if str(type(i).__name__) in __by_name:
```

Comment:
I wonder if it isn't better to compare types directly instead of by name?along the lines of:
```
type(i) in [bpy.types.Object, bpy.types.Mesh, ... ]
```